### PR TITLE
Modify the 100-line code example in the Asterinas Book

### PR DESCRIPTION
**Pull request by dequeueing**
Hi. We were trying to run the 100-line kernel provided in the Gitbook. We found that there were some problems with the original code. So we make some modifications. 

- The hello_world assembly code program. We make some changes but the function is just the same. 
- There were some errors in the original Rust code. We fix them. 

  We have tried that if a scheduler is implemented, then the kernel can work successfully, as shown in the picture below. 

image

  We decide not to add the scheduler in the commit, since it is more than the Asterframe can provide. Plus if the scheduler is added, then entire code would not be close to 100 lines.( unless if you want to change the example to "150-line-example" )

Thank you!

### CODE REVIEW

**Comment by tatetian** [comment]
Thanks for fixing up the example.

And yes, the example is intentionally written to ignore the scheduler. The rationale is that the Asterinas Framework is going to get a major overhaul for its scheduler API. After settling with the ultimate scheduler API, we are going to decide whether an extremely simple, default scheduler will be built into the Asterinas Framework. Then, the 100-line kernel example will be updated accordingly.

**Comment by tatetian** [comment]
@StevenJiang1110 Could you take a look at this PR?

**Review by StevenJiang1110 — COMMENTED** [review]
Thanks for your contribution!

The original assembly code is compiled with nasm, and the new assembly code can be compiled with gcc. gcc is obviously the more common-used compiler.

Counld you please include the compilation command in the doc? Since it may not be trivial.

```sh
gcc -static -nostdlib hello.S -o hello
```

**Review by StevenJiang1110 — APPROVED** [review]
LGTM! Thanks for your contribution!

File: docs/src/framework/a-100-line-kernel.md [hunk 1]
```diff
@@ -7,23 +7,28 @@ we will show a new kernel in about 100 lines of safe Rust.
...
+
+.section .rodata                    # read only data section
+message:
+    .ascii  "Hello, world\n"
+message_end:
+```
 
-section .rodata
-    msg: db "Hello, world!", 10
+Use the following command to compile the program:
```
Comment (tatetian) [inline docs/src/framework/a-100-line-kernel.md:29]: To make the flow of the text more smooth, I rewrite the original sentence by beginning the sentence with the _program_ (the subject before the sentence) and ending it with the _command_ (the subject after the sentence).

> The assembly program above can be compiled with the following command.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**